app/views/nextUpdates/NextUpdatesOptOut.scala.html

```diff
@@ -62,42 +64,69 @@
 	@p() {
 
 		@{
-			optOutViewModel.map {
+			if(showReportingContent){
+				optOutViewModel.map {
 
-				case m:OptOutOneYearViewModel =>
-					if (m.showWarning) {
+					case m:OptOutOneYearViewModel =>
 						Html(
 							s"""
 							<span id="one-year-opt-out-message">
-								${messages("nextUpdates.optOutOneYear-1", m.startYear, m.endYear)}
-								${link(id = Some("single-year-opt-out-warning-link"), link = controllers.optOut.routes.SingleYearOptOutWarningController.show(isAgent).url, messageKey = "nextUpdates.optOutOneYear-2")}
+								${messages("nextUpdates.withReportingFrequencyContent.optOutOneYear-1", m.startYear, m.endYear)}
```

Comment:
think we can simplify this a bit and move some logic out of the views.

We can we use pattern match guards to determine the most of the html and link redirects etc. 

Then just pass like a model of everything into a single

```
Html(
     s"""
            <span id="one-year-opt-out-message">
                 ${messages("nextUpdates.optOutOneYear-1", m.startYear, m.endYear)}
                 ${link(id = Some("single-year-opt-out-warning-link"), link = controllers.optOut.routes.SingleYearOptOutWarningController.show(isAgent).url, messageKey = "nextUpdates.optOutOneYear-2")}
           </span>
      """
)
 ```

Reply:
formatting is a bit of a pain in github comments sorry.